src/components/ui/Badge/Badge.tsx

```diff
@@ -1,26 +1,49 @@
 import { Badge as UIBadge } from '@faststore/ui'
 import type { ReactNode } from 'react'
 import React from 'react'
+import IconButton from 'src/components/ui/IconButton'
+import Icon from 'src/components/ui/Icon'
 
 import './badge.scss'
 
-export type BadgeVariants = 'info' | 'highlighted' | 'neutral'
+export type BadgeVariants = 'info' | 'highlighted' | 'neutral' | 'interactive'
 
 type Props = {
   small?: boolean
   variant?: BadgeVariants
   children: ReactNode
+  onClose?: () => void
 }
 
-const Badge = ({ small = false, variant, children, ...otherProps }: Props) => {
+const Badge = ({
+  small = false,
+  variant,
+  children,
+  onClose,
+  ...otherProps
+}: Props) => {
   return (
     <UIBadge
       className="badge"
       data-store-badge={small ? 'small' : ''}
       data-store-badge-variant={variant}
       {...otherProps}
     >
-      {children}
+      <span>{children}</span>
+      {variant === 'interactive' && (
```

Comment:
Following the @eduardoformiga comment, we can show interactive icon just if the onClose event is pass

```suggestion
      {onClose && (
```